URBNSwiftCarousel/Classes/URBNHorizontalPagedFlowLayout.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  URBNHorizontalPagedFlowLayout.swift
+//  Pods
+//
+//  Created by Kevin Taniguchi on 4/26/16.
+//
+//
+
+import UIKit
+
+/**
+ This is a subclass of UICollectionViewFlowLayout that offsets the cells so that a full cell always appears on the left side.
+*/
+
+public class URBNHorizontalPagedFlowLayout: UICollectionViewFlowLayout {
+    
+    private var minContentOffset: CGFloat {
+        guard let cv = collectionView else { return 0.0 }
+        return -cv.contentInset.left
+    }
+    
+    private var maxContentOffset: CGFloat {
+        guard let cv = collectionView else { return UIScreen.mainScreen().bounds.width }
+        return minContentOffset + cv.contentSize.width - itemSize.width
+    }
+    
+    private var snapStep: CGFloat {
+        return itemSize.width + minimumLineSpacing
+    }
+    
+    //    This offsets the cell's image so that a full image is shown on the left of the collectionview
+    //    Modified from: http://stackoverflow.com/questions/13492037/targetcontentoffsetforproposedcontentoffsetwithscrollingvelocity-without-subcla
+    override public func targetContentOffsetForProposedContentOffset(proposedContentOffset: CGPoint, withScrollingVelocity velocity: CGPoint) -> CGPoint {
+        guard let cv = collectionView else { return CGPointZero }
+        var offSetAdjustment = CGFloat.max
+        var proposedOffSet = proposedContentOffset
+        let targetRect = CGRectMake(proposedContentOffset.x, 0.0, cv.bounds.width, cv.bounds.height)
+        
+        guard let unwrappedAttr = layoutAttributesForElementsInRect(targetRect) else { return CGPointZero }
+        
+        let attr = unwrappedAttr.filter{ $0.representedElementCategory == UICollectionElementCategory.Cell }.first
```

Comment:
I'd like to replace this filter...first pattern with a version of https://github.com/apple/swift-evolution/blob/master/proposals/0032-sequencetype-find.md

we have that in anthro and it could go in convenience. related - I'd like to use the swift layout convenience for setting up vfl constraints and adding subviews